### DISCUSSION TITLE
[@types/papaparse] is not accepting File type for `parse` function

### DISCUSSION THREAD

**Discussion by sirbully** [DefinitelyTyped/DefinitelyTyped#56756] [Issues with a @types package] [answered]
Papaparse `parse` function doesn't allow `File` type as a first parameter. 

Code snippet:
```ts
const parseFile = (file: File) => {
  parse(file, { // file has red squiggly lines here, error below
    header: true,
    transformHeader: (header) => header.toLowerCase(),
    step: function (results) {
      console.log("Row data:", results.data);
      console.log("File progress:", (results.meta.cursor / file.size) * 100);
    }
  });
};
```

Error:
<img width="792" alt="Screen Shot 2021-10-25 at 5 46 59 PM" src="https://user-images.githubusercontent.com/29673385/138673814-54d73d41-d959-4037-9c45-92b3985b4cd4.png">

CodeSandbox link: https://codesandbox.io/s/hardcore-einstein-9mw60?file=/src/App.tsx

Version: 5.3.1

[a comment by an automated account is not shown]

**Comment by sirbully** [comment]
Downgraded to 5.2.6 for now

**Comment by OpportunityLiu** [comment, marked as answer]
You should provide `complete` callback for async parsing.

Missing this callback in async parsing is likely an error.

For ref: https://github.com/typescript-eslint/typescript-eslint/blob/v5.1.0/packages/eslint-plugin/docs/rules/no-floating-promises.md

**Reply by sirbully** [reply]
Hello, indeed adding the complete callback removed the error!
